src/emulator/functionsEmulator.ts

```diff
@@ -513,7 +513,8 @@ export class FunctionsEmulator implements EmulatorInstance {
         projectAlias: this.args.projectAlias,
       };
       const discoveredBuild = await runtimeDelegate.discoverBuild(runtimeConfig, environment);
-      const discoveredBackend = await resolveBackend(discoveredBuild, userEnvOpt, environment);
+      const discoveredBackend = (await resolveBackend(discoveredBuild, userEnvOpt, environment))
```

Comment:
nit* I think this could also work:

```
const { backend: discoveredBackend } = await resolveBackend(discoveredBuild, userEnvOpt, environment)`
```

Not 100% convinced I like this more than just

```
const resolution = await resolveBackend(discoveredBuild, userEnvOpt, environment)`
const discoveredBackend = resolution.backend
```

but `(await foo()).bar` always bothered me a little.

Reply:
+1, tossing it in a variable makes it easier to read.